### DISCUSSION TITLE
Algoritma | SonsuzUs Github

### DISCUSSION THREAD

**Discussion by giscus** [sonsuzus/sonsuzus.github.io#108] [Site]
# Algoritma | SonsuzUs Github

Algoritma Programlama

https://sonsuzus.github.io/posts/algoritma/

**Comment by sonsuzus** [comment]
Kaydırma oyunu programlama.
Yol problemi
Pacman oyunu

**Reply by sonsuzus** [reply]
8 vezir
N vezir

**Reply by sonsuzus** [reply]
Sudoku problemleri
Dama (daha küçük versiyonları)

**Reply by sonsuzus** [reply]
At problemi (tahtayı dolaşan at)

**Comment by ruya-tabiri** [comment]
Mutlu sayı, özel bir matematiksel özellik taşıyan bir sayı türüdür. Bir sayının "mutlu" olup olmadığını belirlemek için aşağıdaki adımlar izlenir:
Sayının her bir rakamının karesini alın ve bu karelerin toplamını hesaplayın.
Elde edilen toplamı yeni bir sayı olarak alın ve bu işlemi tekrarlayın.
Eğer bu süreç sonunda 1 sayısına ulaşırsanız, başlangıçtaki sayı bir mutlu sayıdır.
Eğer bu süreç boyunca aynı sayılar tekrar etmeye başlarsa (bir döngü oluşursa) ve bu sayılar arasında 1 yoksa, başlangıçtaki sayı mutlu sayı değildir.

**Comment by sonsuzus** [comment]
Görseldeki 6 katlı piramidin kaç farklı şekilde doldurulabileceğini bulan kod yazınız. Kurallar :

a) Her satıra [1-9] aralığında sayı yazılabilir.
b) Her satırda kullanılan rakamlar birbirinden farklı olmalı.
c) Her satırda yan yana gelen iki sayı ardışık olmamalı.
d) Zemin kat dışında her kattaki sayı, altında yer alan 2 kutuya yazılmış sayıların ya toplamı ya da farkı olmalı.

![FB_IMG_1706272203828.jpg](https://github.com/sonsuzus/sonsuzus.github.io/assets/52765796/fae013df-e882-4385-9acc-e74cc10b7ddb)

**Comment by sonsuzus** [comment]
Ali ve Veli bir oyun oynamaktadır.

Ali'nin 9 tane 4 yüzlü zarı vardır.
Bu zarların yüzeylerinde (1,2,3,4) vardır. 

Veli'nin 6 tane 6 yüzlü zarı vardır
Bu zarların yüzeylerinde (1,2,3,4,5,6) vardır.

Ali ve Veli ellerindeki zarları atarlar ve kendi zarlarının toplamlarını bulurlar. Toplamı büyük olan oyunu kazanır.

Bu oyunu Ali ve Veli arasında bir milyon kez oynatarak, simülasyon yöntemiyle, Ali'nin kazanma olasılığını yaklaşık olarak bulunuz.

**Comment by sonsuzus** [comment]
Türkçe yazıldığında sessiz harflerinin sayısı, sesli harflerinin sayısının 2 katı olan, rakamları birbirinden farklı en büyük sayıyı bulan kod yazınız.

**Comment by sonsuzus** [comment]
41 sayısı 2+3+5+7+11+13 gibi ardışık asalların toplamı olarak yazılabilen bir asal sayıdır.

Bir milyondan küçük olmak üzere, en uzun ardışık asal sayı zinciri toplamına sahip olan asal sayı hangisidir, bulunuz.

**Comment by sonsuzus** [comment]
197 döngüsel bir asal sayıdır. Çünkü 197, 971, 719 döngüsel permutasyonların hepsi asal sayıdır.

Bir milyondan küçük kaç tane döngüsel asal sayı vardır, bulunuz.

**Comment by sonsuzus** [comment]
Bir sayı alalım. Örneğin 47 ve tersini toplarsak 
47+74=121 
bir palindrom (düz ve ters yazılışı aynı) sayi elde ederiz. 

Bazı palindrom sayilara ulaşmak bu kadar kolay değildir. Örneğin:

349 + 943 = 1292 
1292+2921 = 4213
4213+3124 = 7337

349, üç iterasyon sonunda bir palindrom sayıya ulaştı.

Henüz kanıtlanmamış olsa da 196 gibi bazı sayıların hiçbir zaman palindrom bir sayiya ulaşamayacağı düşünülüyor. Böyle sayılara Lychrel sayıları deniliyor.

İterasyon sayısı limiti olarak 50 alalım.
50 iterasyon sonunda palindrom bir sayıya ulaşamamışsak, muhtemel bir Lychrel sayisiyla başlamışız demektir iterasyona. 

Bu bağlamda, 10.000'den küçük, kaç tane muhtemel Lychrel sayısı vardır, bulunuz.

**Comment by sonsuzus** [comment]
Bir tamsayı dizisi olan sayilar ve bir tamsayı olan hedefSayi verildiğinde, hedefSayi değerini elde etmek için birbirleriyle toplamı hedefSayi olan iki sayının dizi indislerini döndürün.

**Comment by sonsuzus** [comment]
Palindromik sayı 595 ilginçtir çünkü ardışık karelerin toplamı olarak yazılabilir: `6² + 7² + 8² + 9² + 10² + 11² + 12²`

Ardışık kare toplamları olarak yazılabilecek binin altında tam olarak 11 palindrom vardır ve bu palindromların toplamı 4164'tür. Bu problem pozitif tam sayıların kareleriyle ilgili olduğundan `1 = 0²+ 1²` 'nin dahil edilmediğine dikkat edin.

Ardışık karelerin toplamı olarak yazılabilen ve palindromik olan 100,000,000 'dan küçük tüm sayıların toplamını bulun.

**Comment by sonsuzus** [comment]
D(n,r), n pozitif tam sayısı ile başlayıp ardışık r tane pozitif tam sayıdan oluşan bir dizi olsun. ( n > 0, r > 0 )
Verilen sabit bir r değeri ve n = 1,2,3.. için sonsuz tane D(1,r),D(2,r),D(3,r)....D(n,r)... dizileri yazılabilir.
Bir D(n,r) dizisinin elemanları sırasıyla 1,2,3,4...r'nin tam katı olacak şekilde sıralanabiliyorsa D dizisi "n için güzel sıralı" diye adlandırılsın.
r = 36 olsun.
n, 1'den başlamak kaydıyla D(n,36) dizilerinden "n için güzel sıralı" 36. diziyi bulunuz.

**Comment by sonsuzus** [comment]
X sayısının basamaklarının kombinasyonları ile elde edilen tüm sayılar asal ise X döngüsel asaldır. Örneğin 113 sayısı, 113,131 ve 311 sayıları asal olduğu için döngüsel asaldır.

999'dan büyük en küçük döngüsel asal sayı kaçtır?

**Comment by sonsuzus** [comment]
İlk iki haneli sayı *3'ün 1. hanesini değiştirerek, dokuz olası değerden altısının: 13, 23, 43, 53, 73 ve 83’ün tamamının asal olduğu ortaya çıkmaktadır.

56**3 sayısının aynı hanelerinin 3. ve 4. hanelerini aynı rakamla değiştirerek, bu 5 haneli sayı, üretilen on sayı arasında yedi asal sayıya sahip olan ilk örnektir; bu, şunları sağlayarak bir aile oluşturur: 56003, 56113, 56333, 56443, 56663, 56773 ve 56993. Dolayısıyla, bu ailenin ilk üyesi olan 56003, bu özelliğe sahip en küçük asal sayıdır.

Rakamın bir kısmını (mutlaka bitişik haneler olmayabilir) aynı rakamla değiştirerek, sekiz asal sayıyı içeren bir aileye ait olan en küçük asal sayıyı bulun.

**Comment by sonsuzus** [comment]
Damla ve Selin bir zar oyunu oynarlar. İki zarı atacaklardır, 8 gelirse Selin 9 gelirse Damla kazanacaktır. Oyuna Damla başladığına göre hangisinin kazanma olasılığı daha fazladır?

**Comment by sonsuzus** [comment]
Ali ve Betül bir kutudan sırayla şeker alıyorlar. Önce Ali 1 şeker, sonra Betül 2 şeker, sonra Ali 3 şeker, sonra Betül 4 şeker alıyor. Sırası gelen çocuk, kutuda yeterince şeker kalmışsa, bir önceki seferde diğer çocuğun aldığı şeker sayısının bir fazlasını alıyor. Kutuda yeterli sayıda şeker kalmamışsa, sırası gelen çocuk kutudaki bütün şekerleri alıyor. Ali sonuçta 101 şeker almışsa, başlangıçta kutudaki şeker sayısı kaçtır?
